modules/libcom/src/misc/ipAddrToAsciiAsynchronous.h

```diff
@@ -8,6 +8,55 @@
 * in file LICENSE that is included with this distribution.
 \*************************************************************************/
 
+/**
+ * \file ipAddrToAsciiAsynchronous.h
+ * \brief Convert ip addresses to ASCII asynchronously
+ *
+ * ipAddrToAsciiEngine is the first class needed to convert an ipAddr to an
+ * ASCII address.  From the engine, you can use createTransaction() to create a
+ * transaction object.  The transaction object lets you attach callbacks and
+ * convert an address to ASCII.
+ *
+ * Example
+ * -------------
+ *
+ * \code{.cpp}
+ * struct MyResult: ipAddrToAsciiCallBack
+ * {
+ *      virtual void transactionComplete (char const * node) override
+ *      {
+ *          printf("Address is %s\n", node);
+ *      }
+ *
+ *      virtual void show(unsigned level) override const
+ *      {
+ *          printf("This is a MyResult class object.");
+ *      }
+ * };
+ *
+ * ipAddrToAsciiEngine & engine(ipAddrToAsciiEngine::allocate());
+ * ipAddrToAsciiTransaction & trans(engine.createTransaction());
+ *
+ * osiSockAddr addr;
+ * addr.ia.sin_family = AF_INET;
+ * addr.ia.sin_addr.s_addr = htonl(INADDR_LOOPBACK);
+ * addr.ia.sin_port = htons(8080);
+ *
+ * MyResult results;
+ * trans.ipAddrToAscii(addr, results);
+ *
+ * // Do other work and wait for results to finish.
+ * // Note that typically you would use an epicsEvent or similar to coordinate
+ * // making sure the result is ready from your callback.  For this example,
+ * // we'll ignore that complexity and use a naive sleep call instead.
+ * sleep(2);
+ *
+ *
+ * trans.release();
+ * engine.release();
```

Comment:
Updated the example so the callback class manages the transaction object's memory too.  Also added the epicsEvent synchronization rather than the sleep() call.